### PR TITLE
Cosmetic Plando

### PR DESCRIPTION
**Pull request by Cuphat**
I'm going to go ahead and PR this so it doesn't get forgotten, but #1118 should get merged first as this includes those commits.

This adds Cosmetic Plando, changing the cosmetic log into a .json and allowing you to use it to more finely tweak options that aren't exposed by the GUI (and those that are), allowing you to modify any color that would get its own random color, select sound effect options that aren't normally exposed as an option in the GUI, or plan out what music goes where.

Other changes from Dev include splitting the button color options back out to how they currently are on Dev-R (and how they used to be before they were consolidated) as GUI space isn't really a good reason to have them combined and have two versions of the same cosmetic function in the wild. Even with them expanded back out, the UI colors is even with equipment colors and shorter than misc colors, as we can see on Dev-R.

### CODE REVIEW

**Review by Zannick — APPROVED** [review]

File: Cosmetics.py [hunk 1]
```diff
@@ -338,162 +145,387 @@ def patch_navi_colors(rom, settings, log, symbols):
...
+            # Plando
+            if len(plando_colors) > 0 and plando_colors[0].get(trail_part, ''):
+                color = hex_to_color(plando_colors[0][trail_part])
 
-        custom_color = False
-        for index, (address, transparency, white_transparency) in enumerate(sword_trail_addresses):
             # set rainbow option
-            if sword_trail_option == 'Rainbow':
-                rom.write_byte(sword_trail_rainbow_symbol, 0x01)
+            if option_inner == 'Rainbow':
```
Comment (Zannick) [inline Cosmetics.py:373]: Was this supposed to be `option`?

File: Cosmetics.py [hunk 2]
```diff
@@ -338,162 +145,387 @@ def patch_navi_colors(rom, settings, log, symbols):
...
-            else:
-                color = color + [transparency]
+            # grab the color from the list
+            if color is None and option in bombchu_trail_colors:
+                color = list(bombchu_trail_colors[option])
 
-            rom.write_bytes(address, color)
+            # build color from hex code
+            if color is None:
+                color = hex_to_color(option)
```
Comment (Zannick) [inline Cosmetics.py:397]: This is throwing an error for Boomerang when option is 'Yellow'. Should that be a bombchu trail option, or should we check boomerang_trail_colors here as well?

File: Cosmetics.py (hunk 1 above)
Comment (Cuphat) [reply Cosmetics.py]: Nice catch. Fixed.

File: Cosmetics.py (hunk 2 above)
Comment (Cuphat) [reply Cosmetics.py:397]: I pass that in as `trail_color_list`, I guess I just only changed it for random choice and forgot to change it right above those lines to use that instead? Whoops. Fixed, thanks.

Comment (Cuphat) [reply Cosmetics.py:397]: Disregard, I am bad. Going to fix and force push.

Comment (Cuphat) [reply Cosmetics.py:397]: Actually fixed now.

**Pull request merged**